src/System/extract_state.rs

```diff
@@ -21,104 +18,81 @@ pub fn get_state(
     machine: &SystemRecipe,
     system: &TransitionSystemPtr,
 ) -> Result<State, String> {
-    match state_query {
-        QueryExpression::State(loc, clock) => {
-            let mut locations: Vec<&str> = Vec::new();
-
-            for location in loc {
-                match location.as_ref() {
-                    QueryExpression::LocName(name) => locations.push(name),
-                    _ => panic!(),
-                };
-            }
-
-            let locationtuple = build_location_tuple(&locations, machine, system)?;
+    if let QueryExpression::State(loc, clock) = state_query {
+        let mut locations: Vec<&str> = Vec::new();
 
-            let zone = if let Some(clock_constraints) = clock {
-                let mut clocks = HashMap::new();
-                for decl in system.get_decls() {
-                    clocks.extend(decl.clocks.clone());
-                }
-
-                let declarations = Declarations {
-                    ints: HashMap::new(),
-                    clocks,
-                };
-
-                match apply_constraints_to_state(
-                    clock_constraints,
-                    &declarations,
-                    OwnedFederation::universe(system.get_dim()),
-                ) {
-                    Ok(zone) => zone,
-                    Err(wrong_clock) => {
-                        return Err(format!(
-                            "Clock {} does not exist in the transition system",
-                            wrong_clock
-                        ))
-                    }
-                }
-            } else {
-                OwnedFederation::universe(system.get_dim())
+        for location in loc {
+            match location.as_ref() {
+                QueryExpression::LocName(name) => locations.push(name),
+                _ => unreachable!(),
             };
-
-            Ok(State::create(locationtuple, zone))
         }
-        _ => panic!("Expected QueryExpression::State, but got {:?}", state_query),
+
+        Ok(State::create(
+            build_location_tuple(&mut locations.iter(), machine, system)?,
+            create_zone_given_constraints(clock.as_deref(), system)?,
+        ))
+    } else {
+        panic!("Expected QueryExpression::State, but got {:?}", state_query)
```

Comment:
Why not just return `Err`, since the function returns `Result`?

Reply:
@Puvikaran2001 Henrik said to tag u

Reply:
Fixed @t-lohse